ui-modules/utils/quick-launch/quick-launch.js

```diff
@@ -147,23 +144,33 @@ export function quickLaunchDirective() {
         function deployApp() {
             $scope.deploying = true;
 
-            Promise.resolve(buildYaml()).then(appYaml => {
-                quickLaunch.planSender(appYaml)
-                    .then((response) => {
-                        if ($scope.callback) {
-                            $scope.callback.apply({}, [{state: 'SUCCESS', data: response.data}]);
-                        } else {
-                            brSnackbar.create('Application Deployed');
-                        }
-                        $scope.deploying = false;
-                    })
-                    .catch((err) => {
-                        $scope.model.deployError = err.data.message;
-                        $scope.deploying = false;
-                    });
-            });
+            Promise.resolve(buildYaml())
+                .then(appYaml => {
+                    quickLaunch.planSender(appYaml)
+                        .then((response) => {
+                            if ($scope.callback) {
+                                $scope.callback.apply({}, [{state: 'SUCCESS', data: response.data}]);
+                            } else {
+                                brSnackbar.create('Application Deployed');
+                            }
+                            $scope.deploying = false;
+                        })
+                        .catch(({ data }) => {
+                            // handling API error
+                            handleDeployError({ message: data.message });
```

Comment:
Can be just `handleDeployError(data)`?

Reply:
It seems like it could but we're essentially dealing with an http call error and having it mimick a generic `throw 'my message` `Error` object structure. So, it might be best to leave for transparency? I don't mind either way.